docs/INSTALL.md

```diff
@@ -0,0 +1,214 @@
+- [Configuration](#configuration)
+  - [Creating a Product Group](#creating-a-product-group)
+  - [Creating a New Product](#creating-a-new-product)
+  - [Module Settings](#module-settings)
+  - [Configurable Options](#configurable-options)
+- [Adding Existing VPS to Users](#adding-existing-vps-to-users)
+
+## Overview
+
+The UpCloud VPS Module for WHMCS is a versatile tool designed to assist businesses of all sizes in efficiently managing their Cloud Instances. It simplifies the process of configuring and overseeing VM instances, providing users with seamless control and automation capabilities.
+
+## Features
+
+### Admin Features:
+
+1. **Deployment**: Administrators can manually deploy VMs by clicking 'Create' or set up automatic deployment during product setup, specifying location, region, OS, and custom fields like SSH keys or user data (cloud-init) initialization.
+
+2. **Termination**: VM termination can be done manually or automated, with the option to set up automatic termination through WHMCS automation settings, which functions based on WHMCS cron jobs.
+
+3. **Upgrade/Downgrade**: Administrators can manually adjust CPU/RAM/Storage configurations by clicking the 'Change Package' button, or configure automatic upgrades/downgrades.
+
+4. **VM Management**: Administrators can start, stop, restart, suspend, unsuspend VMs, and view all basic VM information including VPS, VNC, IP addresses, etc.
+
+5. **Reverse PTR**: Admins can set and view reverse PTR records.
+
+6. **Login Credentials**: If configured, the system can even send login credentials automatically.
+
+7. **Custom Plans**: Once set by admin during product setup, users can customize their plans and select them for deployment after making a payment.
+
+8. **Cron Job**: The administrator must schedule cron jobs to run every 5 minutes to generate bandwidth usage graphs.
+
+9. **Pricing Adjustment**: While UpCloud has predefined prices for Windows, CPU, RAM, and storage, you have the flexibility to adjust them according to your specific requirements.
+
+### Client Features:
+
+1. **VM Information**: Clients can view basic VM information such as VPS, VNC, IP addresses details.
+
+2. **VNC Management**: Clients can enable/disable VNC access.
+
+3. **Password Management**: Clients can change their VNC passwords.
+
+4. **Reverse PTR**: Clients can view and set reverse PTR records.
+
+5. **VM Control**: Clients can start, stop, and restart their VMs.
+
+6. **Configuration Modification**: Clients can modify server configurations to suit their needs.
+
+7. **Bandwidth Usage Graph**: Clients can view bandwidth usage graphs for intervals spanning 24 hours, a week, a month, and a year.
+
+## Installation
+
+### Downloading and Uploading to WHMCS
+
+1. Obtain the module from our GitHub repository.
+2. Unpack the module on your local system and transfer the "upCloudVps" folder to your WHMCS directory named "/path-to-your-WHMCS/modules/servers/" on your WHMCS installation.
```

Comment:
We may want to adjust this some when we have some actual release zips or the like available. (The zip will likely contain the needed dir structure including parent dirs.)